cloudinit/net/sysconfig.py

```diff
@@ -697,7 +708,16 @@ def _render_vlan_interfaces(cls, network_state, iface_contents, flavor):
                 iface_cfg['ETHERDEVICE'] = iface_name[:iface_name.rfind('.')]
             else:
                 iface_cfg['VLAN'] = True
-                iface_cfg['PHYSDEV'] = iface_name[:iface_name.rfind('.')]
+                iface_cfg.kind = 'vlan'
+
+                rdev = iface['vlan-raw-device']
+                supported = _supported_vlan_names(rdev, iface['vlan_id'])
+                if iface_name not in supported:
+                    LOG.warning(
+                        "Name '%s' not officially supported for vlan "
```

Comment:
"officially supported" by who? This is appearing in the cloud-init log (and we can infer that cloud-init must support network configs with these names, else this would be an error instead of a warning), so we should be explicit about who this is referring to.

(I suspect users don't care about what cloud-init considers officially supported, but they probably are a lot more interested in what their OS vendor considers officially supported. :grin:)